runtime/parachains/src/inclusion.rs

```diff
@@ -155,6 +155,8 @@ decl_error! {
 		InternalError,
 		/// The downward message queue is not processed correctly.
 		IncorrectDownwardMessageHandling,
+		/// At least one upward message sent does not pass the acceptance criteria.
+		InvalidUpwardMessages,
```

Comment:
I imagine it would be helpful for debugging if this identified which particular upward messages were invalid by index, like `InvalidUpwardMessages(Vec<u32>), // vector has indices of invalid messages`.

Reply:
I believe this is a limitation of `decl_error!`.

Reply:
yup, that's right.

Also, as an aside. In runtime engineering we try to make the on-chain runtime code as slim as possible. That's why sometimes we tend to use less diagnostics (you might have noticed attempts at evading `Debug`).

It kinda makes sense, because most of the times, when we run a global consensus, it's not necessarily to bring detailed information. It's inefficient and eats execution time from this super constrained resource called block limit on one hand, and it's not 100% required on the other hand: one can always re-execute a block locally.

Now, it could've been more humane, like if we had something like  https://github.com/paritytech/substrate/issues/2082 or similar, we could potentially split these two cases and get two binaries. But that would require us some work, including to `decl_error` to leverage that.